crates/starknet/src/starknet/dump.rs

```diff
@@ -0,0 +1,178 @@
+                }
+                Transaction::Declare(DeclareTransaction::Version2(tx)) => {
+                    let declare_tx = BroadcastedDeclareTransactionV2::new(
+                        &tx.contract_class,
+                        tx.compiled_class_hash,
+                        tx.sender_address,
+                        tx.max_fee,
+                        &tx.signature,
+                        tx.nonce,
+                        tx.version,
+                    );
+                    self.add_declare_transaction_v2(declare_tx)?;
+                }
+                Transaction::DeployAccount(tx) => {
+                    let deploy_account_tx = BroadcastedDeployAccountTransaction::new(
+                        &tx.constructor_calldata,
+                        tx.max_fee,
+                        &tx.signature,
+                        tx.nonce,
+                        tx.class_hash,
+                        tx.contract_address_salt,
+                        tx.version,
+                    );
+                    self.add_deploy_account_transaction(deploy_account_tx)?;
+                }
+                Transaction::Deploy(_) => return Err(Error::SerializationNotSupported),
+                Transaction::Invoke(InvokeTransaction::Version0(_)) => {
+                    return Err(Error::SerializationNotSupported);
+                }
+                Transaction::Invoke(InvokeTransaction::Version1(tx)) => {
+                    let invoke_tx = BroadcastedInvokeTransaction::new(
+                        tx.sender_address,
+                        tx.max_fee,
+                        &tx.signature,
+                        tx.nonce,
+                        &tx.calldata,
+                        tx.version,
+                    );
+                    self.add_invoke_transaction(invoke_tx)?;
+                }
+                Transaction::L1Handler(_) => return Err(Error::SerializationNotSupported),
+            };
+        }
+
+        Ok(())
+    }
+
+    /// attach starknet transaction to end of existing file
+    pub fn dump_transaction(&self, transaction: &Transaction) -> DevnetResult<()> {
+        match &self.config.dump_path {
+            Some(path) => {
+                let file_path = Path::new(path);
+                if file_path.exists() {
+                    // attach to file
+                    let transaction_dump = serde_json::to_string(transaction).map_err(|_| {
+                        Error::SerializationError { obj_name: "Vec<Transaction>".to_string() }
+                    })?;
+                    let mut file = OpenOptions::new()
+                        .append(true)
+                        .write(true)
+                        .open(file_path)
+                        .map_err(Error::IoError)?;
+                    let lenght = file.seek(SeekFrom::End(0)).map_err(Error::IoError)?;
+                    file.set_len(lenght - 1).map_err(Error::IoError)?; // remove last "]" with set_len
```

Comment:
Should we somehow assert that the last char is indeed `]`? And log an appropriate message if not.

Reply:
Fixed in b57bd62dcc35e0dc81517883b6bc68d13dc85da8